src/Microsoft.Data.Analysis/PrimitiveDataFrameColumn.BinaryOperations.tt

```diff
@@ -54,9 +55,29 @@ namespace Microsoft.Data.Analysis
 <# } #>
         {
 <# if (method.MethodType == MethodType.BinaryScalar) {#>
+<# if (method.IsBitwise == true) { #>
             return <#=method.MethodName#>Implementation(value, inPlace);
+<# } else { #>
+            switch (typeof(U))
+            {
+<# foreach (TypeConfiguration type in typeConfiguration) { #>
+                case Type <#=type.TypeName#>Type when <#=type.TypeName#>Type == typeof(<#=type.TypeName#>):
+                    return <#=method.MethodName#>Implementation(Unsafe.As<U, <#=type.TypeName#>>(ref value), inPlace);
```

Comment:
Had to use Unsafe.As unfortunately. Definitely open to other/better ways.

Reply:
The safe way to do this would probably be something like this (but that boxes):
```C#
return (bool)(object)(value);
```
```suggestion
                    return <#=method.MethodName#>Implementation((<#=type.TypeName#>)(object)(value), inPlace);
```

Reply:
Right, I was trying to avoid that :). Boxing is already showing up in the perf profiles

Reply:
Is it really boxing or is the JIT able to "undo" the boxing?
In most cases JIT is smart enough -- https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACAZQAtsoAHAGW2ADoASgFcAdhgCW+GAG4aNYgGYGZBgGEGAbxoMdDbboDaAKXEYA4jBEwo4sAAoMATw4wIAMzvAIEADYBKPwBdfR1FBi9fBgsMADVsHyEYAB4AFQA+OxSGADd4xL8GAF40hk9vfzsIYAArGDAMP1yEmRoAXyA=